fix init script

daachi opened the pull request:
The pidofproc function should not be redefined here because the same function from the LSB works correctly. Using pgrep will not work correctly if you have a file named similarly to the "daemon-name" open (i.e. "/etc/opt/influxdb/influxdb.conf").

This is with 0.9.2 installed on a base 64-Bit Debian Wheezy system, using the `.deb` from InfluxDB.com.

It's easy to recreate this situation by simply opening the configuration file (i.e. `sudo vi /etc/opt/influxdb/influxdb.conf`), and running `/etc/init.d/influxdb status`. With the configuration file open, `pgrep -f /opt/influxdb/influxd` picks up multiple pids, which causes the init script to report incorrectly that the influxd process is not running (e.g. `influxd Process is not running [ FAILED ]`).

otoolep commented:
Thanks @daachi -- I too noticed this in the script and was wondering why it was there since it is defined elsewhere as you say, but left it for now.

My concern is what will happen on distros that do not have LSB. Is that a concern?

daachi commented:
This is a good point. This commit uses `regex` syntax to check for `influxd` command line, and also checks for the existence of the `pidofproc` function so that we're not overriding it if it exists.

I think this addresses your concerns?

otoolep commented on scripts/init.sh:
```diff
@@ -77,24 +77,26 @@ if [ -r $DEFAULT ]; then
 
-    PID=`pgrep -f $3`
-    local PIDFILE=`cat $2`
+      pid=`pgrep -f '\'$3'\>'`
```
I don't understand this pattern. Can you explain? Why isn't it something like?

`pgrep -f "$#$"` 

The last `$` anchors the process name.

daachi replied:
Your example *should* also work, but I've had issues getting that to work in the past (thought I just tested locally and couldn't get it to work). To be honest I don't remember where I learned this syntax (TLDP, likely), I just know it works. As I'm sure you know, using regex with command line utilities can be... {unreliable, weird, frustrating}.

otoolep replied:
OK, thanks. I'm leaning towards the simpler, clearer option unless you can provide some references. This is an important script obviously, so we need to be careful with what we do.

otoolep commented on scripts/init.sh:
```diff
@@ -77,24 +77,26 @@ if [ -r $DEFAULT ]; then
-    if [ "x$PIDFILE" == "x" ]; then
-        return 1
-    fi
+      if [ "x$pidfile" == "x" ]; then
```
Are these tabs? Can you replace with whitespace?

daachi replied:
These should just be spaces as I'm fairly certain my vim setup doesn't use/strips tabs, but I'll definitely double check.

otoolep commented:
Thanks @daachi -- I do have some questions before we can merge. Also, can you sign the CLA if you have not already done so?

https://influxdb.com/community/cla.html

daachi commented:
I've signed the CLA, for sure.

Also, we're running into some other issues and I will have additional commits related to this. I'd like very much to hear your feedback as well.

daachi commented:
So, otoolep, these commits updates the syntax of the `pgrep` command, as you suggested, corrects indentation, and wraps the remaining functions which can also be found in the LSB.

The problem remains that the `start` function still doesn't work correctly. This can be verified, for example, by opening the config file (from a fresh install) and setting `enabled = true` under the `collectd` section which will render the daemon unable to start. The init script will report incorrectly that the daemon has started.

Before 88307bf2 I had this script working by adding an additional `sleep 5` to the top of the `wait_for_startup` function (which has since been removed). Not elegant, but correct and functional.

I don't know if you will accept this without that fix, but let me know if you want me to fix that as well, or if I should open another PR.

otoolep commented on scripts/init.sh:
```diff
@@ -77,55 +77,63 @@ if [ -r $DEFAULT ]; then
 
-    PID=`pgrep -f $3`
-    local PIDFILE=`cat $2`
+        pid=`pgrep -f "^$3"`
```
Won't this pick up `influxdabc`? Unlikely I know, but again should be anchored on the right it seems?

otoolep commented:
Closed in favour of https://github.com/influxdb/influxdb/pull/4227

Thanks @daachi
